ThirdPartyAdapters/mintegral/mintegral/src/main/java/com/mintegral/mediation/rtb/MintegralRtbBannerAd.java

```
@@ -0,0 +1,191 @@
+  private MediationBannerAdCallback bannerAdCallback;
+  private MBBannerView mbBannerView;
+  private BannerSize bannerSize;
+
+  public MintegralRtbBannerAd(
+          @NonNull MediationBannerAdConfiguration mediationBannerAdConfiguration,
+          @NonNull MediationAdLoadCallback<MediationBannerAd, MediationBannerAdCallback>
+                  mediationAdLoadCallback) {
+    this.adConfiguration = mediationBannerAdConfiguration;
+    this.adLoadCallback = mediationAdLoadCallback;
+  }
+
+  /**
+   * dip convert pixels
+   *
+   * @param context
+   * @param dpValue
+   * @return
+   */
+  private static int dipToPixels(Context context, float dpValue) {
+    final float scale = context.getResources().getDisplayMetrics().density;
+    return (int) (dpValue * scale + 0.5f);
+  }
+
+  private void updateSupportBannerSize() {
+    ArrayList<AdSize> supportedSizes = new ArrayList<>(3);
+    supportedSizes.add(new AdSize(320, 50));
+    supportedSizes.add(new AdSize(300, 250));
+    supportedSizes.add(new AdSize(728, 90));
+    AdSize closestSize = MediationUtils.findClosestSize(adConfiguration.getContext(), adConfiguration.getAdSize(), supportedSizes);
+    if (closestSize == null) {
+      callFailureCallback(MintegralConstants.ERROR_BANNER_SIZE_MISMATCH, "Failed to request banner ad from Mintegral. Invalid banner size.");
+      return;
+    }
+    if (closestSize.equals(AdSize.BANNER)) { // 320 * 50
+      bannerSize = new BannerSize(BannerSize.STANDARD_TYPE, 0, 0);
+    }
+    if (closestSize.equals(AdSize.MEDIUM_RECTANGLE)) { // 300 * 250
+      bannerSize = new BannerSize(BannerSize.MEDIUM_TYPE, 0, 0);
+    }
+    if (closestSize.equals(AdSize.LEADERBOARD)) { // 728 * 90
+      bannerSize = new BannerSize(BannerSize.SMART_TYPE, closestSize.getWidth(), 0);
+    }
+    if (bannerSize == null) {
+      bannerSize = new BannerSize(BannerSize.DEV_SET_TYPE, closestSize.getWidth(), closestSize.getHeight());
+    }
```

Can you add a comment about why we set to `BannerSize.DEV_SET_TYPE` if `closestSize` doesn't match any of the above `AdSize`s?

---

Shouldn't we just return a failure callback if the closest size doesn't match `BANNER`, `MEDIUM_RECTANGLE` and `LEADERBOARD`?

---

There is no best match, we support developers, set custom width and height